packages/client/wallets/smart-wallet/src/smartWalletService.ts

```diff
@@ -0,0 +1,449 @@
+            to: Address;
+            data?: Hex;
+            value?: bigint;
+        },
+        adminSigner: ExternalSigner | PasskeySigner,
+        user: UserParams,
+        chain: SmartWalletChain
+    ): Promise<Hex> {
+        const signerLocator = await this.getSignerLocator(adminSigner);
+        // Create transaction
+        const transactionCreationResponse = await this.crossmintApiService.createTransaction(user, {
+            params: {
+                signer: signerLocator,
+                chain,
+                calls: [
+                    {
+                        to: parameters.to,
+                        value: parameters.value ? parameters.value.toString() : "0",
+                        data: parameters.data ?? "0x",
+                    },
+                ],
+            },
+        });
+        const transactionId = transactionCreationResponse.id;
+
+        // Approve transaction
+        const pendingApprovals = transactionCreationResponse.approvals.pending;
+        if (pendingApprovals.length !== 1) {
+            throw new TransactionApprovalError(`Expected 1 pending approval, got ${pendingApprovals.length}`);
+        }
+        const pendingApproval = pendingApprovals[0];
+        await this.approveTransaction(user, adminSigner, transactionId, pendingApproval.message);
+
+        // Get transaction status until success
+        let transactionResponse: TransactionResponse | null = null;
+        while (transactionResponse === null || transactionResponse.status === "pending") {
+            await sleep(STATUS_POLLING_INTERVAL_MS);
+            transactionResponse = await this.crossmintApiService.getTransaction(user, transactionId);
+        }
+
+        if (transactionResponse.status === "failed") {
+            throw new TransactionFailedError("Transaction sending failed", stringify(transactionResponse.error));
+        }
+
+        // Get transaction hash
+        const transactionHash = transactionResponse.onChain.txId;
+        if (transactionHash === undefined) {
+            throw new TransactionNotFoundError("Transaction hash not found");
+        }
+        return transactionHash;
+    }
+
+    private async createWallet(
+        user: UserParams,
+        signer: ExternalSigner | PasskeySigner
+    ): Promise<CreateWalletResponse> {
+        if ("type" in signer && signer.type === "PASSKEY") {
+            return await this.crossmintApiService.createWallet(user, {
+                type: "evm-smart-wallet",
+                config: {
+                    adminSigner: {
+                        type: "evm-passkey",
+                        id: signer.credential.id,
+                        name: "Default",
+                        publicKey: {
+                            x: `0x${signer.credential.publicKey.x.toString(16)}`,
+                            y: `0x${signer.credential.publicKey.y.toString(16)}`,
+                        },
+                    },
+                    creationSeed: "0",
```

Comment:
Can we remove this parameter for simplicity? I don't think we need it for backwards compatibility since iirc calling `createWallet` here with JWT will return whatever wallet they have, regardless of if it was created with the index or not.

Reply:
makes sense! https://github.com/Crossmint/crossmint-sdk/pull/913/commits/f027d532ba7236b9de901cc9ffd4585c9db47eec